src/librustc_typeck/check/demand.rs

```diff
@@ -246,7 +246,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     // This function checks if the method isn't static and takes other arguments than `self`.
```

Comment:
I am confusing about this comment, is this correct？

Reply:
It's a confusing comment. And the name of the function is also probably wrong. cc @estebank

Reply:
This code seems to only be used for suggestions to convert a type in a whitelist, so I think the "has no input arg" should be read/rewritten as "check that this associated item can be called without any additional arguments, like <&str>::to_string(self)". Feel free to rewrite the docstring and method name.

Reply:
@estebank  @eddyb  Updated, PTAL~  Thanks!